airbyte/_util/temp_files.py

```diff
@@ -21,12 +23,15 @@ def as_temp_files(files_contents: list[dict | str]) -> Generator[list[str], Any,
     """Write the given contents to temporary files and yield the file paths as strings."""
     temp_files: list[Any] = []
     try:
+        temp_dir = OVERRIDE_TEMP_DIR
+
```

Comment:
@coderabbitai and @niyasrad - I think I disagree (respectfully) with the suggestion above.

Looking at the docs for [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile), the default for `dir=None`, which is identical to the behavior when no override is set.

The current implementation is more readable than the suggestion, hence I'd keep what we have presently.

Reply:
I might suggest renaming this variable though - or just mapping the constant directly, for readability.

```python
            temp_file = tempfile.NamedTemporaryFile( # noqa: SIM115 # Avoiding context manager
                mode="w+t",
                delete=False,
                encoding="utf-8",
                dir=OVERRIDE_TEMP_DIR,
                suffix=".json" if use_json else ".txt",
            )
```

Otherwise, you could keep all the same as-is but just rename `temp_dir` to `temp_dir_override` so it is clear to the reader we aren't _always_ passing a non-null value to this arg.